React: update 'render' and 'component' descriptions (#5676)
DevExpress/devextreme-documentation: concepts/50 React Components/40 Component Configuration Syntax/55 Markup Customization/5 Using a Custom Component.md

> You can define custom content in a separate component if you need to utilize state or hooks.

Reviewer: ```suggestion
You can define custom content in a separate component if you need to utilize the state or hooks.
```

> [note] For Class components, we recommend using <a href="https://react.dev/reference/react/PureComponent" target="_blank">`React.PureComponent`</a> because `React.Component` can be re-rendered unnecessarily. Alternatively, you can implement the <a href="https://react.dev/reference/react/Component#shouldcomponentupdate" target="_blank">shouldComponentUpdate()</a> method.

Reviewer 2: ```suggestion
[note] For Class components, we recommend that you use <a href="https://react.dev/reference/react/PureComponent" target="_blank">`React.PureComponent`</a> instead of `React.Component` (the latter can be re-rendered unnecessarily). Alternatively, you can implement a <a href="https://react.dev/reference/react/Component#shouldcomponentupdate" target="_blank">shouldComponentUpdate()</a> method.
```

> The following code demonstrates a [List](/Documentation/Guide/UI_Components/List/Getting_Started_with_List/) which items are rendered in another component. This component is passed to the [itemComponent](/Documentation/ApiReference/UI_Components/dxList/Configuration/#itemComponent) property. 

Reviewer 3: A bit weird statement: "a List wich items are rendered in another component". For me it sounds like the list items are removed from the List and rendered inside another 'master' component (another List or so). But it's not the case. Here I would say that to render List items we create a standalone ListItem component. If required, the ListItem component can be reused in another List.

> In the following code snippet, a standalone ListItem component is created to render [List](/Documentation/Guide/UI_Components/List/Getting_Started_with_List/) items. If required, the ListItem component can be reused in another List. This component is passed to the [itemComponent](/Documentation/ApiReference/UI_Components/dxList/Configuration/#itemComponent) property. 

Reviewer 2: ```suggestion
In the following code snippet, a standalone ListItem component is created to render [List](/Documentation/Guide/UI_Components/List/Getting_Started_with_List/) items. This component specifies the [itemComponent](/Documentation/ApiReference/UI_Components/dxList/Configuration/#itemComponent) property value.  If necessary, you can reuse this ListItem component in another List. 
```

> You can define custom content markup in a separate component. This component is rendered in the virtual DOM, and any changes trigger a rerender of the component.

Reviewer 4: > any changes trigger a rerender of the component

It's unclear which "any" changes trigger re-render (sounds scary). Let's use language from the first .md file here.

> For example, you need to display a dynamic form in your application, and each time a different form configuration is required. You can implement a component for your form and pass it to the *'component'* property.

Reviewer 4: > each time a different form configuration is required

Unclear what do we mean by "time". Render? Some data refresh?
